### PR TITLE
Add gNOI-4.1: Software Upgrade test

### PR DESCRIPTION
**Pull request by bstoll**
This adds test coverage for GNOI OS.Install, OS.Activate, and OS.Verify RPCs.

### CODE REVIEW

**Review by hariramnat — DISMISSED** [review]

File: feature/system/tests/osinstall/osinstall_test.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# gNOI-4.1: Software Upgrade
```
Comment (liulk) [inline feature/system/tests/osinstall/osinstall_test.md:1]: Aukur suggested that this be called README.md so they're easier to locate.

File: feature/system/tests/osinstall/osinstall_test.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Summary
+
+Validate new software can be copied and activated on single and dual supervisors.
+
+## Procedure
```
Comment (liulk) [inline feature/system/tests/osinstall/osinstall_test.md:7]: Just want to double check, did the test implement all the procedure here? The missing items should have a TODO in this file.

**Review by liulk — CHANGES_REQUESTED** [review]
Sorry I forgot to request change. See my previous comments.

File: feature/system/tests/osinstall/osinstall_test.md (hunk 2 above)
Comment (bstoll) [reply feature/system/tests/osinstall/osinstall_test.md]: Yes - there is one item I added a TODO for due to partial coverage but this should implement the test procedure.

**Review by liulk — APPROVED** [review]

**Pull request merged**